### ISSUE TITLE
Train with dedicated test set

### ISSUE THREAD

**Issue by tmbo** [RasaHQ/rasa_core#289]
Allow a user to specify a separate story data set when training that is used for the evaluation when training. This should replace the automatic splitting used by keras to determin the validation dataset. If no test set is passed, the current behaviour should be used (splitting the training data set and evaluate on parts of it).

**Comment by tmbo** [comment]
@amn41 how important do you think is this from a machine learning perspective? Just took a look at it, it is possible but makes things more complicated. What would be the added benefit instead of using the splitting keras does? I mean you can still evaluate the trained model on a separate test set.

**Comment by amn41** [comment]
this isn't super pressing. It's more for if you have specific validation / test stories you want to use (and keep separate from training). I think useful for us for research purposes but not important enough if it requires big changes, we can stick to automatic splitting
